### ISSUE TITLE
[RFC]: Using Op.* as functions on top of keys

### ISSUE THREAD

**Issue by ephys** [sequelize/meetings#13]
### Is your feature request related to a problem? Please describe.

I've personally never liked the ergonomics of having to use sequelize operators as keys for an object.
The following code

```javascript
const where = { id: { [Op.notIn]: blockList } };
```

feels easier to read when written like this:

```javascript
const where = { id: Op.notIn(blockList) };
```

### Describe the solution you'd like

We could make it possible to use Op.* as both object keys and functions using toPrimitive. That would make both snippets of code above valid.

Proof of concept:

```javascript
const symbolNotIn = Symbol('notIn');

function notIn(val) {
  return { [symbolNotIn]: val };
}

notIn[Symbol.toPrimitive] = () => {
  return symbolNotIn;
}

console.log(notIn([1, 2]));        // output: { [Symbol(notIn)]: [1, 2] }
console.log({ [notIn]: [1, 2] });  // output: { [Symbol(notIn)]: [1, 2] }

export const Op = {
  // ...
  notIn,
  // ...
};
```

### Why should this be in Sequelize

I believe this would make the usage of operators easier to read and understand.

### Describe alternatives/workarounds you've considered

If using Symbol.toPrimitive feels too much like magic, those functions could be provided somewhere else. Sequelize.and, Sequelize.or, etc already exist.

### Additional context

N/A

### Is this feature dialect-specific?

- [x] No. This issue is relevant to Sequelize as a whole.
- [ ] Yes. This issue only applies to the following dialect(s): XXX, YYY, ZZZ

### Would you be willing to implement this feature by submitting a Pull Request?

- [x] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by ephys** [comment]
Due to TypeScript not supporting `Symbol.toPrimitive`, maybe we could go the same route as `Sequelize.and` / `Sequelize.or` and expose a utility function for most / all operators:

```
import { notEqual } from 'sequelize';

User.findAll({
  where: { id: notEqual(5) }
});
```

**Comment by sdepold** [comment]
In general I like the proposal but would maybe not expose it via sequelize but maybe add a layer or nesting?

Import { Operators} from sequelize 

Const { notEqual}=Operators

?

**Comment by ephys** [comment]
I agree, but `Operators` is very close to `Op` even though one is a namespace for functions, and the other one for symbols. I fear people might start using `Operators` as keys by accident

If we expose something like `Operators`, I think we should move `Sequelize.and` & `Sequelize.or` over to `Operators` too

This issue is frustrating. I have a solution that is almost working but it causes typescript to stop reporting errors with the current way of using operators:

![image](https://user-images.githubusercontent.com/1280915/152952704-bc13ece4-0a88-4b00-b1f8-9bd42a03c119.png)

**Comment by WikiRik** [comment]
Feel free to make a new issue that supersedes this one to have a cleaner discussion
